modules/geo/src/main/java/org/opensearch/geo/search/aggregations/metrics/GeoBoundsAggregator.java

```diff
@@ -51,22 +47,11 @@
 import java.util.Map;
 
 /**
- * Aggregate all docs into a geographic bounds
+ * Aggregate all docs into a geographic bounds for field GeoPoint.
  *
  * @opensearch.internal
  */
-final class GeoBoundsAggregator extends MetricsAggregator {
-
-    static final ParseField WRAP_LONGITUDE_FIELD = new ParseField("wrap_longitude");
-
-    private final ValuesSource.GeoPoint valuesSource;
-    private final boolean wrapLongitude;
-    DoubleArray tops;
-    DoubleArray bottoms;
-    DoubleArray posLefts;
-    DoubleArray posRights;
-    DoubleArray negLefts;
-    DoubleArray negRights;
+final class GeoBoundsAggregator extends AbstractGeoBoundsAggregator<ValuesSource.GeoPoint> {
```

Comment:
Will there be a new `ValuesSource.GeoShape` when shape docvalues are added to lucene? Or are you planning on creating a `ValuesSource.Geo` that handles both?

Reply:
I planning to create a new ValueSource which will be ValueSource.GeoShape for geoshapes. Now I think about it, may be we can combine them form and single coresource value type. I might need to check.

But for this PR I don't want to change it. I think I will take it in another PR as this PR is getting long already.

I will do it in the followup PR, as there we will be a lot of classes that will be required to be changed.